doc/ikfast/ikfast_tutorial.rst

```diff
@@ -73,210 +53,79 @@ Inside your catkin workspace's ``./src`` directory: ::
-
- export IKFAST_PRECISION="5"
- cp "$MYROBOT_NAME".dae "$MYROBOT_NAME".backup.dae  # create a backup of your full precision dae.
- rosrun moveit_kinematics round_collada_numbers.py "$MYROBOT_NAME".dae "$MYROBOT_NAME".dae "$IKFAST_PRECISION"
-
-From experience we recommend 5 decimal places, but if the OpenRAVE IKFast generator takes to long to find a solution, lowering the number of decimal places should help.
-
-To see the links in your newly generated Collada file
-
-You may need to install package **libsoqt4-dev** to have the display working: ::
-
- openrave-robot.py "$MYROBOT_NAME".dae --info links
-
-This is useful if you have a 7-dof arm and you need to fill in a ``--freeindex`` parameter, discussed later.
-
-To test your newly generated Collada file in OpenRAVE: ::
-
- openrave "$MYROBOT_NAME".dae
-
-You should see your robot.
-
-.. image:: openrave_panda.png
-   :width: 700px
-
-Create IKFast Solution CPP File
--------------------------------
-Once you have a numerically rounded Collada file its time to generate the C++ .h header file that contains the analytical IK solution for your robot.
+  rosrun xacro xacro -o $MYROBOT_NAME.urdf $MYROBOT_NAME.urdf.xacro
 
 Select IK Type
 ^^^^^^^^^^^^^^
-You need to choose which sort of IK you want. See `this page <http://openrave.org/docs/latest_stable/openravepy/ikfast/#ik-types>`_ for more info.  The most common IK type is *transform6d*.
+You need to choose which type of IK you want to solve for. See `this page <http://openrave.org/docs/latest_stable/openravepy/ikfast/#ik-types>`_ for more info.
+The most common IK type is *transform6d*.
 
 Choose Planning Group
 ^^^^^^^^^^^^^^^^^^^^^
-If your robot has more than one arm or "planning group" that you want to generate an IKFast solution for, choose one to generate first. The following instructions will assume you have chosen one ``<planning_group_name>`` that you will create a plugin for. Once you have verified that the plugin works, repeat the following instructions for any other planning groups you have. For example, you might have 2 planning groups: ::
+If your robot has more than one arm or "planning group" that you want to generate an IKFast solver for, you need to repeat the following process for each group.
+The following instructions will assume you have chosen one ``<planning_group_name>``. Furthermore, you need to know the names of the base link and the endeffector link of the chain to solve for.
 
- <planning_group_name> = "left_arm"
- <planning_group_name> = "right_arm"
+Generate IKFast MoveIt plugin
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-To make it easy to use copy/paste for the rest of this tutorial. Set a ``PLANNING_GROUP`` environment variable. eg: ::
+To generate the IKFast MoveIt plugin, issue the following command: ::
 
- export PLANNING_GROUP="panda_arm"
-
-Identify Link Numbers
```

Comment:
If we can preserve the existing method while also supporting the auto ikfast script, then we can disregard this comment

Reply:
This section is not needed with the new script anymore. You directly specify the base and eef link _names_ and these should be clear to any user.

Reply:
It may be 'prudent' to explain/keep around the link id part. For unknown reasons the ikfast generators can sometimes get confused when using names, and link ids have saved me once or twice in those situations.

Reply:
@gavanderhoorn, Do you want to add a proposal for this paragraph?

Reply:
Tbh I'm really overloaded right now, but just wanted to relay my experiences so they could potentially be taken into account.

sorry :(

Reply:
if we just keep the old explanation of how to output the link names, this should be resolved.